src/main/java/org/jabref/logic/database/DatabaseMerger.java

```diff
@@ -119,7 +119,8 @@ private void mergeGroups(MetaData target, MetaData other, String otherFilename,
                             "Imported " + newGroupName,
                             GroupHierarchyType.INDEPENDENT,
                             keywordDelimiter);
-                    newGroups.setGroup(group);
+                    newGroups.setGroup(group, false, false,
```

Comment:
Add code comment for null. Why is null passed. – Readers should blt need to read JavaDoc of the called method to understand.

Reply:
Maybe pass List.of()?

Reply:
> Add code comment for null. Why is null passed. – Readers should blt need to read JavaDoc of the called method to understand.

```
public List<FieldChange> setGroup(AbstractGroup newGroup, boolean shouldKeepPreviousAssignments,
                                      boolean shouldRemovePreviousAssignments, List<BibEntry> entriesInDatabase) {
        AbstractGroup oldGroup = getGroup();
        groupProperty.set(Objects.requireNonNull(newGroup));

        List<FieldChange> changes = new ArrayList<>();
        boolean shouldRemoveFromOldGroup = shouldRemovePreviousAssignments && (oldGroup instanceof GroupEntryChanger);
        boolean shouldAddToNewGroup = shouldKeepPreviousAssignments && (newGroup instanceof GroupEntryChanger);
        if (shouldAddToNewGroup || shouldRemoveFromOldGroup) {...}
return changes;
```

Since the old code only input the group, then it never enter the `if` part, so there is no need to pass the `entriesInDatabase` parameter, I just keep the same input like org.jabref.gui.collab.groupchange.GroupChange#applyChange

Reply:
My comment in other words: `null` should be avoided. 

Maybe the `@deprecated` flag was more than just a one-line fix with the thinking, we need to do know.

I checked the code. The `setGroup` with more parameters has a different semantics.

Undo you change - and remove `@deprecated` annoation.

The method makes sence in the context. With your additons, the method use is WTF!

Reply:
True, when I changed it I also think it is WTF!